src/Sdk/DTWebApi/WebApi/TimelineRecord.cs

```diff
@@ -1,78 +1,86 @@
+                this.PercentComplete = recordToBeCloned.PercentComplete;
+                this.RecordType = recordToBeCloned.RecordType;
+                this.Result = recordToBeCloned.Result;
+                this.ResultCode = recordToBeCloned.ResultCode;
+                this.StartTime = recordToBeCloned.StartTime;
+                this.State = recordToBeCloned.State;
+                this.TimelineId = recordToBeCloned.TimelineId;
+                this.WorkerName = recordToBeCloned.WorkerName;
+                this.RefName = recordToBeCloned.RefName;
+                this.ErrorCount = recordToBeCloned.ErrorCount;
+                this.WarningCount = recordToBeCloned.WarningCount;
+                this.NoticeCount = recordToBeCloned.NoticeCount;
+                this.AgentPlatform = recordToBeCloned.AgentPlatform;
+
+                if (recordToBeCloned.Log != null)
                 {
-                    Id = recordToBeCloned.Log.Id,
-                    Location = recordToBeCloned.Log.Location,
-                };
-            }
+                    this.Log = new TaskLogReference
+                    {
+                        Id = recordToBeCloned.Log.Id,
+                        Location = recordToBeCloned.Log.Location,
+                    };
+                }
 
-            if (recordToBeCloned.Details != null)
-            {
-                this.Details = new TimelineReference
+                if (recordToBeCloned.Details != null)
                 {
-                    ChangeId = recordToBeCloned.Details.ChangeId,
-                    Id = recordToBeCloned.Details.Id,
-                    Location = recordToBeCloned.Details.Location,
-                };
-            }
+                    this.Details = new TimelineReference
+                    {
+                        ChangeId = recordToBeCloned.Details.ChangeId,
+                        Id = recordToBeCloned.Details.Id,
+                        Location = recordToBeCloned.Details.Location,
+                    };
+                }
 
-            if (recordToBeCloned.m_issues?.Count> 0)
-            {
-                this.Issues.AddRange(recordToBeCloned.Issues.Select(i => i.Clone()));
-            }
+                if (recordToBeCloned.m_issues?.Count > 0)
+                {
+                    this.Issues.AddRange(recordToBeCloned.Issues.Select(i => i.Clone()));
+                }
 
-            if (recordToBeCloned.m_previousAttempts?.Count > 0)
-            {
-                this.PreviousAttempts.AddRange(recordToBeCloned.PreviousAttempts);
-            }
+                if (recordToBeCloned.m_previousAttempts?.Count > 0)
+                {
+                    this.m_previousAttempts.AddRange(recordToBeCloned.m_previousAttempts);
+                }
 
-            if (recordToBeCloned.m_variables?.Count > 0)
-            {
-                this.m_variables = recordToBeCloned.Variables.ToDictionary(k => k.Key, v => v.Value.Clone());
```

Comment:
This was a bug.  The original might have had a case-insensitive Variables Dictionary, but the clone's would be case-sensitive.